Sources/VaporJWT/AuthError.swift

```diff
@@ -0,0 +1,89 @@
+/// Errors thrown during the parsing, creation
+/// and verification of JWT tokens by this package.
+public struct AuthError: Swift.Error {
+    public let code: Int
+    public let reasonPhrase: String
+    public let origin: Swift.Error?
+
+    init(code: Int, reasonPhrase: String, origin: Swift.Error?) {
+        self.code = code
+        self.reasonPhrase = reasonPhrase
+        self.origin = origin
+    }
+
+    public static func noAuthorizationHeader(_ origin: Swift.Error? = nil) -> Error {
```

Comment:
@tannernelson have we verified that works to allow an undefined `Error` type to be directly compared with another undefined or concrete `Error` type?

That seems like Swift wouldn't support that in current state, I think we should throw it in a test or something. I believe it'd have to return the concrete type in these functions. (Not positive)